mantidproject/mantid
# Updating naming in new backend

**Description of work.**

This updates the naming in the new GUI to match scientific specifications. After consulting the scientists the following naming scheme was decided upon:
* If no user specified name is provided the name should consist of {RunNumber}{Detector}_{Dimensionality}_{WavelengthRange}{PhiRange}_{TimeSlice}
* If a user specified name is provided the output workspaces are given this name. Suffixes are appended in the case of merged, all, or sliced reductions in order to distinguish the different workspaces.
* Also as part of this PR the input data is grouped and the transmission workspaces are grouped.
* The wavelength range over which the transition workspace is fitted is also adjusted. 

**To test:**
A good place to start testing would be:
* Get the data from [loqdemo.zip](https://github.com/mantidproject/mantid/files/2286736/loqdemo.zip) and insure the directory is added to the Mantid search paths.
* Then run the following script
```
from sans.command_interface.ISISCommandInterface import *

MASKFILE = FileFinder.getFullPath('MaskFile.txt')
BATCHFILE = FileFinder.getFullPath('batch_mode_reduction.csv')
Clean()
LOQ()
MaskFile(MASKFILE)
SetEventSlices('3:2:9')
BatchReduce(BATCHFILE, format='.nxs', saveAlgs={"SaveCanSAS1D": ".xml"}, combineDet='merged')
```

Check that 5 workspace groups are created once the script has finished running. Check that these make sense and hopefully look like the image below.

![image](https://user-images.githubusercontent.com/32058966/44089566-22849e30-9fbf-11e8-9f0c-4d562a18c4fd.png)

* You can modify the batch file in order to test in the case of a non user specified name. Simply open it in excel or similar and remove the entry entitled 'first_time'



Fixes #23215 




---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
